src/browser-frontend/sagas/server-event-handlers.js

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+
+import SharedActions from '../../shared/actions/shared-actions';
+import PagesModelActions from '../actions/pages-model-actions';
+import * as DomainPagesSelectors from '../selectors/domain-pages-selectors';
+import * as UIPagesSelectors from '../selectors/ui-pages-selectors';
+
+function* handleCmdOrControlQ({ meta: client }) {
```

Comment:
Note: there's a fair bit of plumbing going on for this to happen. The request to register keyboard shortcuts is sent from the browser frontend to the browser server, which pipes it onward to the browser runner. When keys are pressed, the event bubbles back the same route: runner -> server -> frontend. This implementation is in line with the proposed architecture in #31, following a "single two-lane street" principle where all communication between all components follows a single path/bus, instead of allowing all components to communicate with each other as they would like. In principle, this should reduce complexity in the long run.